components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -800,6 +800,22 @@ void AdsServiceImpl::OnResetAllState(
   VLOG(1) << "Successfully reset ads state";
 }
 
+void AdsServiceImpl::GetHardwareInfo() {
+  base::SysInfo::GetHardwareInfo(base::BindOnce(
+      &AdsServiceImpl::OnGetHardwareInfo, base::Unretained(this)));
+}
+
+void AdsServiceImpl::OnGetHardwareInfo(
+    base::SysInfo::HardwareInfo hardware) {
+  ads::SysInfoPtr sys_info = ads::SysInfo::New();
+  sys_info->manufacturer = hardware.manufacturer;
+  sys_info->model = hardware.model;
+
+  bat_ads_service_->SetSysInfo(std::move(sys_info), base::NullCallback());
```

Comment:
This can't work, you're calling `SetSysInfo` before bat_ads_service is created https://github.com/brave/brave-core/pull/7446/files#diff-aea813da741d9093d0388c807fd40741c5ca24ff03c80aaeaebdd3944465f44bR841